transfer_queue/storage/clients/ray_storage_client.py

```
@@ -0,0 +1,126 @@
+from typing import Any
+
+import ray
+import torch
+from torch import Tensor
+
+from transfer_queue.storage.clients.base import TransferQueueStorageKVClient
+from transfer_queue.storage.clients.factory import StorageClientFactory
+
+@ray.remote
+class RayGpuObjectRefStorage:
+    def __init__(self):
+        self.storage_dict = {}
+
+    def put_gpu_obj_ref(self, keys: list[str], gpu_obj_refs: list[ray.ObjectRef]):
+        for key, gpu_obj_ref in zip(keys, gpu_obj_refs):
+            self.storage_dict[key] = gpu_obj_ref
+
+    def get_gpu_obj_ref(self, keys: list[str]) -> list[ray.ObjectRef]:
+        obj_refs = [self.storage_dict.get(key, None) for key in keys]
+        return obj_refs
+
+    def clear_gpu_obj_ref(self, keys: list[str]):
+        for key in keys:
+            if key in self.storage_dict:
+                del self.storage_dict[key]
+
+@StorageClientFactory.register("RDT")
+class RayStorageClient(TransferQueueStorageKVClient):
+
+    def __init__(self, config: dict[str, Any]):  
+        if not ray.is_initialized():  
```

`ray.init()` is neccessary. However, this check should probably not be performed by the storage client; it might be the responsibility of the user or the upper-level interface.

---

Done.

---

a tricky thing here: ray driver does not support `ray.put(v,  _tensor_transport="nixl")` . so if ray is inited here, this process becomes a driver. The following `put()` will fail. So I prefer to raise an error here